WordPress/Classes/Services/SignupService.swift

```diff
@@ -1,84 +1,74 @@
-                                        let account = service.createOrUpdateAccount(withUsername: username, authToken: bearer_token)
-                                        if service.defaultWordPressComAccount() == nil {
-                                            service.setDefaultWordPressComAccount(account)
-                                        }
-
-                                        let createdAccount = (responseDictionary?[ResponseKeys.createdAccount] as? Int ?? 0) == 1
-                                        if createdAccount {
-                                            defer {
-                                                WPAppAnalytics.track(.createdAccount, withProperties: ["source": "google"])
-                                            }
-                                            success(createdAccount)
-                                        } else {
-                                            defer {
-                                                WPAppAnalytics.track(.signupSocialToLogin)
-                                            }
-                                            // we need to sync the blogs for existing accounts to be able to display the Login Epilogue
-                                            BlogSyncFacade().syncBlogs(for: account, success: {
-                                                success(createdAccount)
-                                            }, failure: { (_) in
-                                                // the blog sync failed but the user is already logged in
-                                                success(createdAccount)
-                                            })
-                                        }
-                                    },
-                                    failure: failure)
+        remote.createWPComAccount(withGoogle: googleToken,
+                                  andLocale: locale,
+                                  andClientID: configuration.wpcomClientId,
+                                  andClientSecret: configuration.wpcomSecret,
+                                  success: { response in
+
+            guard let username = response?[ResponseKeys.username] as? String,
+                let bearer_token = response?[ResponseKeys.bearerToken] as? String else {
+                    failure(ServiceError.unknown)
+                    return
+            }
+
+            let createdAccount = (response?[ResponseKeys.createdAccount] as? Int ?? 0) == 1
+            success(createdAccount, username, bearer_token)
+        }, failure: { error in
+            failure(error ?? ServiceError.unknown)
+        })
     }
+}
 
 
+// MARK: - Private
+//
+private extension SignupService {
 
-    // MARK: Private Instance Methods
-
-    @objc func anonymousApi() -> WordPressComRestApi {
-        return WordPressComRestApi(userAgent: WPUserAgent.wordPress())
+    private var anonymousAPI: WordPressComRestApi {
```

Comment:
is this private redundant, given it's already in a `private extension`?

Reply:
Reasoning behind the double private is:

```
class Test {
    func doSomething() {
        accessible()
        inaccessible() // This should break
    }
}

private extension Test {

    func accessible() {   }

    private func inaccessible() { }
}
```

This was **TRUE** the last time i've checked (Swift 4.0). Now... in the latest compiler, the code posted above doesn't break anymore. So yeah, let's nuke the double private, since ... the compiler changed it's behavior!